aya-build/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+use std::{
+    env, fs,
+    io::{BufRead as _, BufReader},
+    path::PathBuf,
+    process::{Child, Command, Stdio},
+};
+
+// Re-export `cargo_metadata` to having to encode the version downstream and risk mismatches.
+pub use cargo_metadata;
+use cargo_metadata::{Artifact, CompilerMessage, Message, Package, Target};
+
+/// Build binary artifacts produced by `packages`.
+///
+/// This would be better expressed as one or more [artifact-dependencies][bindeps] but issues such
+/// as:
+///
+/// * <https://github.com/rust-lang/cargo/issues/12374>
+/// * <https://github.com/rust-lang/cargo/issues/12375>
+/// * <https://github.com/rust-lang/cargo/issues/12385>
+///
+/// prevent their use for the time being.
+///
+/// [bindeps]: https://doc.rust-lang.org/nightly/cargo/reference/unstable.html?highlight=feature#artifact-dependencies
+pub fn build_ebpf(packages: impl IntoIterator<Item = Package>) {
```

Comment:
I would prefer returning an error instead of panicking.

I know that 99% of users will unwrap in build.rs anyways (that's fine and expected), but still, I would stick to the principle that libraries should return an error. There always might be that 1% of people who want to create some more sophisticated xtask and handle the error somehow.

Reply:
Will do. In the meantime can you review #1098 which is a prerequisite of this?